# Análise e resultados

## Passo a passo

# Primeiros passos
- Transformar Medu e Fedu (nível de escolaridade da mãe e do pai) de STRING para INTEGER
- none = 0
- primary education(quarta série) = 1
- quinta a nona série = 2
- secondary education = 3
- higher education = 4

## Calcular média dos três periodos (G1,G2 e G3) em média geral
## Media geral = 11


In [5]:
import pandas as pd


resultados = pd.DataFrame({
    'Escola': ['GP', 'MS', 'TOTAL'],
    'Total de alunos': [420, 228, 648],
    'Acima da média': [321, 104, 425],
    '% acima da média': ['76.4%', '45.6%', '65.6%'],
    'Abaixo da média': [95, 120, 215],
    '% abaixo da média': ['22.6%', '52.6%', '33.1%'],
})


resultados


Escola  Total de alunos  Acima da média % acima da média  Abaixo da média  \
0     GP              420             321            76.4%               95   
1     MS              228             104            45.6%              120   
2  TOTAL              648             425            65.6%              215   

  % abaixo da média  
0             22.6%  
1             52.6%  
2             33.1%

### 📊 Número e porcentagem de reprovados por escola e genêro dos alunos

In [6]:
import pandas as pd

# Tabela final com análise
data = [
    ['GP', 'F', 21, '50.0%', '22.6%'],
    ['GP', 'M', 21, '50.0%', '22.6%'], 
    ['MS', 'F', 30, '58.8%', '32.3%'],
    ['MS', 'M', 21, '41.2%', '22.6%'],
    ['Total geral', '', 93, '100.0%', '100.0%']
]

columns = ['school', 'sex', 'failures', '% por escola', '% do total']
df_final = pd.DataFrame(data, columns=columns)

print("🎯 DISTRIBUIÇÃO DE ALUNOS COM REPROVAÇÕES")
print("="*60)
print(df_final.to_string(index=False))


🎯 DISTRIBUIÇÃO DE ALUNOS COM REPROVAÇÕES
     school sex  failures % por escola % do total
         GP   F        21        50.0%      22.6%
         GP   M        21        50.0%      22.6%
         MS   F        30        58.8%      32.3%
         MS   M        21        41.2%      22.6%
Total geral            93       100.0%     100.0%


In [7]:
import pandas as pd

# Criando a tabela dinâmica
data = [
    ['no', 35, 30, 65, '46.15%'],
    ['yes', 512, 63, 575, '10.96%'],
    ['Missing data', '-', '-', 2, '-'],
    ['TOTAL', 547, 93, 640, '14.53%']
]

columns = ['Higher Education', 'No Failures', 'With Failures', 'Total', '% with Failures']
df_he = pd.DataFrame(data, columns=columns)

# Exibindo a tabela
print("🎓 Relação entre Interesse em Ensino Superior e Reprovações")
print("="*70)
df_he

🎓 Relação entre Interesse em Ensino Superior e Reprovações


Higher Education No Failures With Failures  Total % with Failures
0               no          35            30     65          46.15%
1              yes         512            63    575          10.96%
2     Missing data           -             -      2               -
3            TOTAL         547            93    640          14.53%

## 💡 Análise das Porcentagens

### 📈 Distribuição das 93 Reprovações:

**🏫 Por Escola:**
- GP: 42 alunos (45.2%)
- MS: 51 alunos (54.8%)

**👥 Por Gênero:**
- Mulheres: 51 alunas (54.8%)
- Homens: 42 alunos (45.2%)

**🎯 Padrões Específicos:**
- GP: Distribuição perfeita (50%-50%)
- MS: Mais mulheres reprovadas (58.8% vs 41.2%)

**📊 Em Relação ao Total de Alunos:**
- 14.3% dos alunos têm histórico de reprovação
- MS tem taxa ligeiramente maior de reprovações

### 📈 Diferença Expressiva:
- **Interessados em ensino superior**: Apenas 10,96% têm reprovações  
- **Desinteressados em ensino superior**: 46,15% têm reprovações
- **Gap de 35,19 pontos percentuais!**

### 🎯 Impacto Significativo:
- Alunos sem interesse em faculdade têm **4,2x mais chances** de reprovação
- **575 alunos** (89,8%) planejam ensino superior vs **65** (10,2%) que não planejam

### 💡 Conclusão Forte:
O interesse em continuar os estudos é um **forte indicador** de desempenho acadêmico atual!

In [20]:
import pandas as pd

# Dados para análise
print("🔍 ANÁLISE DO GRUPO DE ALTO RISCO")
print("="*55)

total_sem_internet = 151  # Do dataset anterior
total_com_reprovacoes = 93  # Do dataset anterior
grupo_risco = 36

print(f"\n📊 CONTEXTO GERAL:")
print(f"   • Total de alunos sem internet: {total_sem_internet}")
print(f"   • Total de alunos com reprovações: {total_com_reprovacoes}")
print(f"   • Alunos SEM internet E COM reprovações: {grupo_risco}")
print(f"   • Representa {grupo_risco/total_sem_internet*100:.1f}% dos sem internet")
print(f"   • Representa {grupo_risco/total_com_reprovacoes*100:.1f}% dos reprovados")

print(f"\n🏫 DISTRIBUIÇÃO POR ESCOLA:")
print(f"   • GP: {12} alunos em risco → {12/77*100:.1f}% dos sem internet da GP")
print(f"   • MS: {24} alunos em risco → {24/74*100:.1f}% dos sem internet da MS")

print(f"\n👥 DISTRIBUIÇÃO POR GÊNERO:")
print(f"   • Mulheres: {16} em risco → {16/88*100:.1f}% das sem internet")
print(f"   • Homens: {20} em risco → {20/63*100:.1f}% dos sem internet")

print(f"\n🎯 PROBABILIDADE DE RISCO:")
print(f"   • Se não tem internet: {grupo_risco/total_sem_internet*100:.1f}% chance de reprovar")
print(f"   • Se tem internet: {(93-36)/490*100:.1f}% chance de reprovar")


🔍 ANÁLISE DO GRUPO DE ALTO RISCO

📊 CONTEXTO GERAL:
   • Total de alunos sem internet: 151
   • Total de alunos com reprovações: 93
   • Alunos SEM internet E COM reprovações: 36
   • Representa 23.8% dos sem internet
   • Representa 38.7% dos reprovados

🏫 DISTRIBUIÇÃO POR ESCOLA:
   • GP: 12 alunos em risco → 15.6% dos sem internet da GP
   • MS: 24 alunos em risco → 32.4% dos sem internet da MS

👥 DISTRIBUIÇÃO POR GÊNERO:
   • Mulheres: 16 em risco → 18.2% das sem internet
   • Homens: 20 em risco → 31.7% dos sem internet

🎯 PROBABILIDADE DE RISCO:
   • Se não tem internet: 23.8% chance de reprovar
   • Se tem internet: 11.6% chance de reprovar


### 🎯 Habitos dos alunos

In [8]:
import pandas as pd

# Criando DataFrame com formatação
data = [
    ['Female study time below avg (hours)', 2.19, 1.88, 2.19],
    ['Male study time below avg (hours)', 1.8, 1.58, 1.8],
    ['Female study time above avg (hours)', 2.19, 1.88, 2.19],
    ['Male study time above avg (hours)', 1.8, 1.58, 1.8],
    ['Female go out above avg (hours)', 3.13, 3.15, 3.13],
    ['Male go out above avg (hours)', 3.19, 3.49, 3.19],
    ['Female go out below avg (hours)', 3.13, 3.15, 3.13],
    ['Male go out below avg (hours)', 3.19, 3.44, 3.19]
]

columns = ['Metric', 'GP', 'MS', 'Total']
df = pd.DataFrame(data, columns=columns)

# Formatação para melhor visualização com 2 casas decimais
styled_df = df.style.set_properties(
    **{'text-align': 'left'}
).set_table_styles([
    {'selector': 'th', 'props': [('text-align', 'left')]}
]).format({
    'GP': '{:.2f}',
    'MS': '{:.2f}', 
    'Total': '{:.2f}'
})

print("📊 Comportamento por Gênero: Estudo vs Saídas em Horas")
print("="*65)
styled_df


📊 Comportamento por Gênero: Estudo vs Saídas em Horas


## 💡 Principais Observações

### 📚 Padrões de Estudo:
- **👩 Mulheres estudam mais** que homens em ambas escolas
- **🏫 GP tem maior tempo de estudo** que MS
- **⏰ Tempo constante** independente de estar acima/abaixo da média

### 🎉 Padrões de Saídas:
- **👨 Homens saem mais** que mulheres, especialmente em MS
- **🏫 MS tem valores mais altos** em saídas
- **📊 Pouca variação** entre categorias de alunos que estão acima/abaixo da média

### 🎯 Insight Interessante:
Comportamentos são consistentes independente de estarem acima ou abaixo da média, sugerindo padrões culturais/gênero bem estabelecidos.

### 💡 Correlação de consumo de alcool (dalc), faltas(absences) e saídas (goout) com o coeficiente de failure

In [9]:
import pandas as pd

# Criando a tabela dinâmica
data = [
    ['GP', 1.46, 0.15, 4.16, 3.15],
    ['MS', 1.56, 0.31, 2.55, 3.25],
    ['Total geral', 1.49, 0.21, 3.60, 3.19]
]

columns = ['school', 'AVERAGE de Dalc', 'AVERAGE de failures', 'AVERAGE de absences', 'AVERAGE de goout']
df_correlacao = pd.DataFrame(data, columns=columns)

# Exibindo a tabela
print("📊 Tabela Dinâmica - Correlação entre Variáveis")
print("="*65)
df_correlacao

📊 Tabela Dinâmica - Correlação entre Variáveis


school  AVERAGE de Dalc  AVERAGE de failures  AVERAGE de absences  \
0           GP             1.46                 0.15                 4.16   
1           MS             1.56                 0.31                 2.55   
2  Total geral             1.49                 0.21                 3.60   

   AVERAGE de goout  
0              3.15  
1              3.25  
2              3.19

📈 Correlações Positivas com Reprovações:
- **🍷 Consumo de álcool (Dalc)**: MS (1,56) > GP (1,46) → MS tem mais reprovações
- **🎉 Frequência de saídas (Goout)**: MS (3,25) > GP (3,15) → MS tem mais reprovações

### 🔄 Correlação Inversa/Inesperada:
- **❌ Faltas (Absences)**: GP (4,16) > MS (2,55) → Porém GP tem MENOS reprovações
  - *Possível explicação: faltas justificadas ou outros fatores intervenientes*

### 🎯 Conclusão:
Variáveis comportamentais (Dalc, Goout) mostram forte correlação com desempenho acadêmico, enquanto faltas apresentam relação contraintuitiva que merece investigação adicional.

 ### 💡 Desempenho escolar a partir do nível de educação da mãe

In [10]:
print ('==Desempenho escolar dos alunos por escola relacionado ao nível de educação da mãe==')
import pandas as pd

# Criando a tabela formatada
data = [
    [0, 'GP', '0.00%'],
    [0, 'MS', '25.00%'],
    ['0', 'Total', '16.67%'],
    [1, 'GP', '29.55%'],
    [1, 'MS', '39.51%'],
    ['1', 'Total', '34.32%'],
    [2, 'GP', '18.75%'],
    [2, 'MS', '30.00%'],
    ['2', 'Total', '23.08%'],
    [3, 'GP', '6.06%'],
    [3, 'MS', '22.58%'],
    ['3', 'Total', '10.00%'],
    [4, 'GP', '7.07%'],
    [4, 'MS', '17.86%'],
    ['4', 'Total', '9.45%'],
    ['Total geral', '', '20.63%']
]

columns = ['Fedu', 'school', 'AVERAGE de failures']
df_final = pd.DataFrame(data, columns=columns)

# Exibindo a tabela
df_final.style.hide(axis='index').set_properties(**{'text-align': 'left'})

==Desempenho escolar dos alunos por escola relacionado ao nível de educação da mãe==


### 💡 Desempenho escolar dos alunos por escola relacionados ao nível de educação do pai

In [11]:
print ('==Desempenho escolar dos alunos por escola relacionado ao nível de educação do pai==')
import pandas as pd

# Criando a tabela dinâmica do nível de escolaridade dos pais
data = [
    ['GP', 0, '66.67%'],
    ['GP', 1, '33.93%'],
    ['GP', 2, '18.70%'],
    ['GP', 3, '16.00%'],
    ['GP', 4, '2.24%'],
    ['GP', 'Total', '15.14%'],
    ['MS', 0, '0.00%'],
    ['MS', 1, '39.76%'],
    ['MS', 2, '22.58%'],
    ['MS', 3, '47.22%'],
    ['MS', 4, '12.50%'],
    ['MS', 'Total', '30.80%'],
    ['Total geral', '', '20.63%']
]

columns = ['school', 'Medu', 'AVERAGE de failures']
df_medu = pd.DataFrame(data, columns=columns)

# Exibindo a tabela
print("Tabela Dinâmica - AVERAGE de failures por school e Medu")
print("="*55)
df_medu

==Desempenho escolar dos alunos por escola relacionado ao nível de educação do pai==
Tabela Dinâmica - AVERAGE de failures por school e Medu


school   Medu AVERAGE de failures
0            GP      0              66.67%
1            GP      1              33.93%
2            GP      2              18.70%
3            GP      3              16.00%
4            GP      4               2.24%
5            GP  Total              15.14%
6            MS      0               0.00%
7            MS      1              39.76%
8            MS      2              22.58%
9            MS      3              47.22%
10           MS      4              12.50%
11           MS  Total              30.80%
12  Total geral                     20.63%

# 💡Principais Insights

 ### Escola GP tem 423 alunos e Escola MS conta com 226 alunos, totalizando 649. Porém, 9 alunos possuem dados de idade acima dos dos 19 anos, como 21 e 22, o que provavelmente significa um erro na produção da tabela, já que eles não estão mais em idade escolar.

1️⃣ Escola GP é mais forte

    76.4% dos alunos acima da média vs 45.6% em MS
    Diferença significativa

2️⃣ Falhas concentram em alunos fracos

    Abaixo da média: ~55% com falhas
    Acima da média: ~3.5% com falhas
    Diferença: ~15x mais

3️⃣ Padrão consistente

    14.3% dos alunos têm histórico de reprovação
    Colégio MS tem uma taxa ligeiramente maior de reprovações (51 alunos) e mais mulheres reprovadas do que homens (58.8% vs 41.2%)
    Colégio GP tem menor taxa de reprovação (42) e apresenta paridade entre homens e mulheres reprovadas (50% a 50%)
   

    4️⃣ - Nivel de ensino dos pais afetam levemente a taxa de sucesso dos filhos

     O nível de ensino das mães tem um nível maior de impacto no nível de sucesso dos filhos
     Filhos de pais e mães com nível 1 de escolaridade (até a quarta série) são os que mais repetiram de ano, muito mais do que aqueles sem escolaridade nenhuma (nível 0).
     # Ponto de investigação: Alunos de pais com nível de escolaridade nível 2 (quinta a nona série) estão performando muito mal; 43,7% estão perdendo de ano
     
5️⃣ - Número 5

    Enquanto consumo de alcool e "saídas" (goout) impactam a taxa de perde de ano, principalmente na escola MS, a taxa de falta a aula (absence) tem relação inversa: os estudantes da escola GP foram os que mais faltaram em média, mas aqueles que menos repetiram de ano.

    
6️⃣ - Número 6

Alunos sem interesse em ingressar no ensino superior tem 4,2x mais chances de serem reprovados



🚨 Recomendações Baseadas nos Dados
 

1️⃣ **Foco Preventivo**: Intervir nos alunos sem interesse em ensino superior

2️⃣ **Programa de Orientação**: Mostrar conexão entre desempenho atual e futuro acadêmico 

3️⃣ **Acompanhamento**: Monitorar especialmente os 65 alunos do grupo de risco (65 alunos que não tem interesse em nível superior e compoem a maior taxa de reprovação)

4️⃣ **Motivação**: Desenvolver programas que conectem sucesso escolar com oportunidades futuras

7️⃣ - **Familia**: Entender porque alunos de pais com nível 3 de ensino no colégio MS tem uma altissíma taxa de reprovação

8️⃣ - **Escolas** Colégio MS necessita fazer uma analise interna e melhorar o ensino. A escola MS precisa de intervenção URGENTE e ESPECÍFICA, focando em:

    🎓 Famílias com ensino médio (nível 3 educacional)

    👨 Homens com excesso de saídas e álcool

    👩 Mulheres com baixo rendimento

    📚 Cultura de estudo menos desenvolvida

**Prioridade MÁXIMA: Grupo de pais com ensino médio (47.22% reprovação) - este é o insight mais valioso e específico da MS!** 

9️⃣ - Falta de acesso a internet

   Programa emergencial de acesso à internet para MS, aulas de reforço presenciais para alunos sem internet e materiais didáticos offline para este grupo
 
 MS concentra 67% dos casos (24 de 36 alunos)
 

🎓 Conclusões

✅ Performance e falhas são altamente correlacionadas

✅ GP tem desempenho superior

✅ Oportunidade: ~55% dos alunos abaixo da média repetiram de ano → intervenção possível

✅ No geral das duas escolas, 93 de 640 alunos reprovaram, ou seja, 14,5%. É um número aceitavel, mas que pode ser melhorado

✅Destes, 51 foram da escola MS e 42 foram da escola GP. GP conta com 423 alunos e 12,06% de taxa de reprovação, enquanto MS conta com 226 alunos e 22,57* de taxa de reprovação.

✅AUSÊNCIA DE INTERNET é fortemente correlacionada com REPROVAÇÕES
Chance de reprovar sem internet: 23.8%

 Chance de reprovar com internet: 11.6%

 🔥 2.1x MAIS chances sem internet

 36 alunos estão na situação de MAIOR risco: sem internet + reprovações
 
 Homens representam 56% do grupo de risco

